app/models/admin_setting.rb

```diff
@@ -38,14 +39,17 @@ class << self
   end
 
   def self.default_skin
-    settings = current
-    if settings.default_skin_id.present?
-      Rails.cache.fetch("admin_default_skin") { settings.default_skin }
+    if self.first
+      self.first.default_skin_id ? self.first.default_skin : Rails.cache.fetch(default_skin_cache_key) { Skin.default }
     else
-      Skin.default
+      Rails.cache.fetch(default_skin_cache_key) { Skin.default }
```

Comment:
#3382 introduced caching for `Skin.default`, so wrapping it in another cache block is a bit redundant (both here and above on line 43).

In general, I'm not sure what the changes to this function are intended to accomplish. Currently, I think you're wrapping a cache block around something that's already cached, and deleting a cache block around something that's not otherwise cached. Is there something subtly wrong with the current caching set-up that I'm missing? Or did the changes just get obsoleted by #3382?

Reply:
Yeah, #3382 added caching for the default skin, so the changes in this PR are no longer needed. I'll close this.

I'll keep [AO3-4466](https://otwarchive.atlassian.net/browse/AO3-4466) open though, because @zz9pzza's comment from the issue is still true:

> If you are logged in there is still one look up, so better but not completely finished.
>
>     ↳ app/helpers/skins_helper.rb:25:in `set_site_skin'
>     Skin Load (1.3ms) SELECT `skins`.* FROM `skins` WHERE `skins`.`id` = 1 LIMIT 1
>     ↳ app/helpers/skins_helper.rb:26:in `set_site_skin'
>     Cache read: views//v1/site_skin_1_2016-08-14 04:41:15 -0400
>     Cache read: skins_generation/site_skin